src/Symfony/Component/Console/Command/Command.php

```diff
@@ -628,4 +633,46 @@ private function validateName($name)
             throw new InvalidArgumentException(sprintf('Command name "%s" is invalid.', $name));
         }
     }
+
+    /**
+     * Locks a command.
+     *
+     * @param bool $silent
+     *
+     * @return bool
+     *
+     * @throws RuntimeException
+     */
+    protected function lock($silent = false)
+    {
+        if (!class_exists('Symfony\Component\Filesystem\LockHandler')) {
+            throw new RuntimeException('To enable the locking feature you must install the symfony/filesystem component');
+        }
+
+        $this->lock = new LockHandler($this->getName());
+
+        if (!$this->lock->lock()) {
```

Comment:
The `LockHandler::lock` method has a parameter for blocking mode, which can't be used this way.